app/secrets.py

```diff
@@ -8,7 +10,9 @@
 ]
 
 
-def validate_required_secrets(secrets, additional_required_secrets=None):
+def validate_required_secrets(
+    secrets: Mapping[str, str], additional_required_secrets: Optional[list[str]] = None
```

Comment:
1. `secrets` is a Mapping of Mapping which has string key values.

```js
{
   "secrets": {
 		"some-key": "some-value",
	}
}
```

2. `additional_required_secrets` is not optional from our current usage. Line 16 would throw an error if it was actually `None` as you can `+` a list and None. Also, we should use `Sequence` over `list` as per type hint style guide, abstract instead of concrete for arguments.